sql/core/src/test/scala/org/apache/spark/sql/execution/command/PlanResolutionSuite.scala

```diff
@@ -0,0 +1,257 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.command
+
+import java.net.URI
+
+import org.apache.spark.sql.{AnalysisException, SaveMode}
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.catalyst.analysis.AnalysisTest
+import org.apache.spark.sql.catalyst.catalog.{BucketSpec, CatalogStorageFormat, CatalogTable, CatalogTableType}
+import org.apache.spark.sql.catalyst.parser.CatalystSqlParser
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.execution.datasources.{CreateTable, DataSourceResolution}
+import org.apache.spark.sql.types.{IntegerType, StringType, StructType}
+
+class PlanResolutionSuite extends AnalysisTest {
+  import CatalystSqlParser._
+
+  def parseAndResolve(query: String): LogicalPlan = {
+    DataSourceResolution(conf).apply(parsePlan(query))
+  }
+
+  private def extractTableDesc(sql: String): (CatalogTable, Boolean) = {
+    parseAndResolve(sql).collect {
+      case CreateTable(tableDesc, mode, _) => (tableDesc, mode == SaveMode.Ignore)
+    }.head
+  }
+
+  test("create table - with partitioned by") {
```

Comment:
why are some test cases duplicated?

Reply:
These were split into two test suites:

* A DDL suite that validates SQL conversion to the parsed logical plan
* A DDL/Resolution suite that runs the exact same tests to validate that the extra conversion step doesn't change anything

Reply:
shall we merge these 2 test suites? Or we need to add document to say that, when adding a new DDL command, we need to add a test in both `DDLParserSuite` and `PlanResolutionSuite` for different purpose.

Reply:
shall we merge these 2 test suites? Or we need to add document to say that, when adding a new DDL command, we need to add a test in both `DDLParserSuite` and `PlanResolutionSuite` for different purpose.

Reply:
No.

These suites are in separate modules. The parser suite is in catalyst, where new tests will go. The old tests that validate there are no changes when converting SQL -> parsed plan -> datasource plan must be in core because the plans don't exist in catalyst.

There is no need to add documentation because these will evolve separately. New parsed plans for DDL will be added to the DDL suite. Conversions will be tested when they are added. For example, we are going to add DSv2 plans for create and CTAS. Those conversions will be tested independently.